core_gems.rb

```diff
@@ -10,5 +10,5 @@
 download "oj", "3.3.10"
 download "tzinfo", "1.2.2"
 download "tzinfo-data", "1.2016.5"
-download "google-protobuf", "3.5.1"
-download "grpc", "1.8.3"
+download "google-protobuf", "gem-fetch-3.5.1"
+download "grpc", "gem-fetch-1.8.3"
```

Comment:
Broader question here, instead of prefixing the version with gem-fetch, could we instead change "download" to "gem-fetch"? If we could, that would clean up all of the version parsing you're doing now.

Reply:
This is doable (well, as `gem_fetch`, because Ruby doesn't support hyphens in identifiers), but gets us deeper into the `td-agent` fork.
One possible way would be to change `gems_parser.rb` as follows:
```ruby
  def download(name, ver)
    @target_files << [name, ver, :download]
  end

  def gem_fetch(name, ver)
    @target_files << [name, ver, :fetch]
  end
```
and then actually use (switch on) the `an` argument in the loop over `gp.target_files` in `gem_downloader`.

Reply:
Do you think we should explore this option, or that what we have with the version prefix is just as clear?

Reply:
I like this proposal, but not if it turns into a rathole. @qingling128, do you want to try this quickly, and, if you can't get it working in ~30 min, stay with the current approach?
Syntactically, we could probably also make something like this work:
```ruby
download "grpc", "1.8.3", :with_native_extensions
```
Using the same approach to pass the info through to `gem_downloader`. I don't have a preference.

Reply:
Good point. Switched to `fetch "grpc", "1.8.3"` and cleaned up the string matching logic.

I'm leaning towards this format instead of having a third `:with_native_extensions` param for readability.